pkg/limatmpl/github.go

```diff
@@ -0,0 +1,196 @@
+}
+
+// getGitHubDefaultBranch queries the GitHub API to get the default branch for a repository.
+func getGitHubDefaultBranch(ctx context.Context, org, repo string) (string, error) {
+	apiURL := fmt.Sprintf("https://api.github.com/repos/%s/%s", org, repo)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, http.NoBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("User-Agent", "lima")
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+
+	// Check for GitHub token in environment for authenticated requests (higher rate limit)
+	token := cmp.Or(os.Getenv("GH_TOKEN"), os.Getenv("GITHUB_TOKEN"))
+	if token != "" {
+		req.Header.Set("Authorization", "token "+token)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to query GitHub API: %w", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read GitHub API response: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("GitHub API returned status %d: %s", resp.StatusCode, string(body))
+	}
+
+	var repoData struct {
+		DefaultBranch string `json:"default_branch"`
+	}
+
+	if err := json.Unmarshal(body, &repoData); err != nil {
+		return "", fmt.Errorf("failed to parse GitHub API response: %w", err)
+	}
+
+	if repoData.DefaultBranch == "" {
+		return "", fmt.Errorf("repository %s/%s has no default branch", org, repo)
+	}
+
+	return repoData.DefaultBranch, nil
+}
+
+// resolveGitHubSymlink checks if a file at the given path is a symlink/redirect to another file.
+// If the file contains a single line without YAML content, it's treated as a path to the actual file.
+// Returns the redirect path if found, or the original path otherwise.
+func resolveGitHubSymlink(ctx context.Context, org, repo, branch, filePath string) (string, error) {
+	url := fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", org, repo, branch, filePath)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("User-Agent", "lima")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch file: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("file not found or inaccessible: status %d", resp.StatusCode)
+	}
+
+	// Read first 1KB to check the file content
+	buf := make([]byte, 1024)
+	n, err := resp.Body.Read(buf)
+	if err != nil && !errors.Is(err, io.EOF) {
+		return "", fmt.Errorf("failed to read file content: %w", err)
+	}
+	content := string(buf[:n])
+
+	if LooksLikeSymlink(content) {
+		redirectPath := strings.TrimSpace(content)
+		if redirectPath != "" {
+			return redirectPath, nil
```

Comment:
This is not correct if the `filePath` includes directories because the `redirectPath` will be relative to this directory and not to the repo root.

Reply:
Fixed now:

```console
❯ tree -a -I .git ~/git/jandubois
/Users/jan/git/jandubois
├── .lima.yaml -> templates/demo.yaml
├── docs
│   └── lima.yaml -> ../templates/demo.yaml
└── templates
    └── demo.yaml

3 directories, 3 files

❯ l tmpl url github:jandubois//docs/lima.yaml
WARN[0000] GitHub locator "github:jandubois//docs/lima.yaml" replaced with "https://raw.githubusercontent.com/jandubois/jandubois/main/templates/demo.yaml" is still EXPERIMENTAL
https://raw.githubusercontent.com/jandubois/jandubois/main/templates/demo.yaml
```